### PR TITLE
Track dropped metrics due to series limits too

### PR DESCRIPTION
**Pull request by gouthamve**
We alert on our customers hitting limits but they don't trigger when the sample is dropped due to series limits being hit.

### CODE REVIEW

File: pkg/ingester/user_state.go [hunk 1]
```diff
@@ -188,6 +188,7 @@ func (u *userState) getSeries(metric labelPairs) (model.Fingerprint, *memorySeri
 	// serially), and the overshoot in allowed series would be minimal.
 	if u.fpToSeries.length() >= u.limits.MaxSeriesPerUser(u.userID) {
 		u.fpLocker.Unlock(fp)
+		validation.DiscardedSamples.WithLabelValues("rate_limited", u.userID).Inc()
```
Comment (tomwilkie) [inline pkg/ingester/user_state.go]: Can you put a const for the label in the validation package pls?

Comment (tomwilkie) [reply pkg/ingester/user_state.go]: In fact there is already one in the distributor, so unify the two please.

**Comment by tomwilkie** [comment]
LGTM with one nit.

**Comment by gouthamve** [comment]
@tomwilkie Done

**Comment by tomwilkie** [comment]
Just need to add a comment to appease the linter now.

**Comment by bboreham** [comment]
Why were all three cases given the same label?  This makes it harder to track issues.

**Pull request merged**